src/components/navigation/header/ComparisonItem.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { FC } from 'react';
+import { useI18n } from '@smg-automotive/i18n-pkg';
+import { chakra } from '@chakra-ui/react';
+
+import { CompareIcon } from 'src/components/icons';
+import Count from 'src/components/count';
+import Box from 'src/components/box';
+import Badge from 'src/components/badge';
+
+export const shouldShowComparisonLink = (
+  comparisonItemIds?: number[] | null,
+): comparisonItemIds is number[] => {
+  return !!comparisonItemIds && Array.isArray(comparisonItemIds);
+};
+
+export const getComparisonUrl = (comparisonItemIds: number[]) => {
+  const baseUrl = 'comparison';
+  if (comparisonItemIds.length === 0) return baseUrl;
+  return `${baseUrl}/${comparisonItemIds.join('/')}`;
+};
+
+type Props = {
+  comparisonItemIds?: number[] | null;
+};
+
+const ComparisonItem: FC<Props> = ({ comparisonItemIds }) => {
+  const { t, language } = useI18n();
+
+  return (
+    <chakra.a
+      position="relative"
+      href={`/${language}/${getComparisonUrl(comparisonItemIds ?? [])}`}
+      display={
+        shouldShowComparisonLink(comparisonItemIds)
```

Comment:
i need to use CSS for the visibility. Otherwise you'll get a hydration errror

Reply:
when I returned null, I got an hydration error from the server side rendering that the client HTML does not match the server HTML. Or do you mean something different?

Reply:
No, I thought it it's null and then changes it would not result in a hydration error as the condition changes client side...